src/app/store/country.selector.ts

```diff
@@ -9,7 +9,24 @@ export const selectAllCountries = createSelector(
   (state: CountryState) => state.countries
 );
 
-export const selectSelectedCountry = createSelector(
-  selectCountryState,
-  (state: CountryState) => state.selectedCountry
-);
+// Selector to get a selected country by name
+export const selectCountryByName = (countryName: string) =>
+  createSelector(selectAllCountries, (countries) =>
+    countries.find((country) => country.name.common === countryName)
+  );
+
+// Selector to filter countries based on a search term
+export const selectFilteredCountries = (
+  searchTerm: string,
+  searchBy: 'name' | 'capital'
+) =>
+  createSelector(selectAllCountries, (countries) => {
+    if (!searchTerm) {
+      return countries;
+    }
+    return countries.filter((country) =>
+      searchBy === 'name'
+        ? country.name.common.toLowerCase().includes(searchTerm.toLowerCase())
+        : country.capital?.[0].toLowerCase().includes(searchTerm.toLowerCase())
```

Comment:
No filteration needs to happen on the frontend.. Please call the api's for filtering.